docs/_guide/properties.md

```diff
@@ -28,158 +78,232 @@ static get properties() {
 
-* An object with two function properties, `fromAttribute` and `toAttribute`. `fromAttribute` performs deserialization, and `toAttribute` performs serialization:
+  * To **observe** an attribute (set a property from an attribute), the attribute value must be converted from a string to match the property type.
 
-  ```js
-  propName: { type: {
-    toAttribute: serializerFunction,
-    fromAttribute: deserializerFunction
-  }}
-  ```
+  * To **reflect** an attribute (set an attribute from a property), the property value must be converted to a string.
 
-By default, `type`, `fromAttribute` and `toAttribute` are the `String` constructor. 
+LitElement defines a **converter** for every declared property. A converter is an object that contains two functions:
 
-To handle deserialization of strings, numbers, and booleans, you can use the corresponding constructor:
+  * `fromAttribute` converts from an [observed attribute](#observed-attributes) to a property when the attribute changes.
+  * `toAttribute` converts from a property to its [reflected attribute](#reflected-attributes) when the property changes.
+
+#### Use the default converter {#conversion-type}
+
+LitElement has a default converter which handles `String`, `Number`, `Boolean`, `Array`, and `Object` property types.
+
+To use the default converter, specify the `type` option in your property declaration:
 
 ```js
-return { 
-  myString: { type: String },
-  myNumber: { type: Number },
-  myBoolean: { type: Boolean }
-}
+// Use LitElement's default converter 
+prop1: { type: String },
+prop2: { type: Number },
+prop3: { type: Boolean },
+prop4: { type: Array },
+prop5: { type: Object }
 ```
 
-Note that when a property of type `Boolean` is deserialized, if it is truthy, the corresponding attribute is created. If the property is falsy, the attribute is removed.
+The table below shows how the default converter handles conversion for each type.
 
-#### Serialize and deserialize object properties
+For all types: 
 
-**Objects (including arrays) must be handled differently.** LitElement has no default handling for converting between object properties and string attributes. If you need to serialize and deserialize complex properties, you must implement a `type` for them.
+* If `toAttribute` returns `null`, the attribute is removed. 
+* If `toAttribute` returns `undefined`, the attribute is not changed.
 
-One option is to use `JSON.parse` and `JSON.stringify`:
+|**Type**|**`fromAttribute` returns**|**`toAttribute` returns**|
+|---|---|---|
+| `String` | Attribute value | Property value |
+| `Number` | Number(attributeValue) | Property value |
+| `Boolean` | `true` if non-`null`, `false` otherwise | `''` if truthy, `null` otherwise |
+| `Array` | `JSON.parse(attributeValue)` | `null` if the property value is `null`;  `JSON.stringify(propertyValue)` otherwise |
```

Comment:
Actually, fo objects and arrays `toAttribute` returns "`null` (attribute is removed) if the property value is `null` _or_ `undefined`".

Reply:
Could you expand on that point? I somehow thought the property value being undefined wouldn't affect the attribute. Will try to find where I got that impression (could have been code comments or comments on another issue)

Reply:
Er never mind that, i guess it would be bad to `JSON.stringify(undefined)`...